dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.ts

```diff
@@ -0,0 +1,76 @@
+import Big from 'big.js';
+import { differenceInDays, isBefore, startOfMonth } from 'date-fns';
+import orderBy from 'lodash.orderby';
+import { MonthAndYear, ReportPeriod, TfmFacilityAmendment } from '@ukef/dtfs2-common';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { NotFoundError } from '../../../../../errors';
+import { convertTimestampToDate } from '../../../../../helpers';
+
+const getLatestCompletedAmendmentCoverEndDate = (amendments: TfmFacilityAmendment[]): number | undefined => {
+  const completedAmendments = amendments.filter(({ status }) => status === 'Completed');
+  const latestAmendmentWithDefinedCoverEndDate = orderBy(completedAmendments, ['updatedAt'], ['desc']).find(({ coverEndDate }) => coverEndDate);
+  return latestAmendmentWithDefinedCoverEndDate?.coverEndDate ?? undefined;
+};
+
+const getLatestTfmFacilityValues = async (
```

Comment:
I'm worried this is slightly hidden if e.g. what Andy highlighted about new types amendments coming in actually comes to fruition. Is having latest values for these things relevant in other parts of the codebase? If so, I wonder if we can write a more common helper that then might get updated for free if new amendments exist...

Reply:
In matching the way we have helpers for fee records, I have added a `tfm-facilities.helper.ts` to the helpers directory which then gets the latest cover end date. In terms of where else this happens in the codebase, I believe this kind of logic is usually performed in `portal-api` and/or `trade-finance-manager-api` before it is sent to the respective UI server which is why this commonised logic doesn't really exist in `dtfs-central-api`

Reply:
Hmm I see.. `findLatestCompletedAmendment` in tfm-api seems vaguely like something one would want in the common libs perhaps and then to use but tbh that does something a little weird that I'm not sure I follow, almost implying perhaps that a later amendment contains all the previous amendment fields not being amended again?